janitor/utils.py

```diff
@@ -1671,3 +1677,170 @@ def _computations_as_categorical(df: pd.DataFrame, **kwargs) -> pd.DataFrame:
+            columns_to_select.start,
+            columns_to_select.stop,
+            columns_to_select.step,
+        )
+        start_check = any((start is None, isinstance(start, str)))
+        stop_check = any((stop is None, isinstance(stop, str)))
+        step_check = any((step is None, isinstance(step, int)))
+        if not start_check:
+            raise ValueError(
+                """
+                The start value for the slice
+                must either be a string or `None`.
+                """
+            )
+        if not stop_check:
+            raise ValueError(
+                """
+                The stop value for the slice
+                must either be a string or `None`.
+                """
+            )
+        if not step_check:
+            raise ValueError(
+                """
+                The step value for the slice
+                must either be an integer or `None`.
+                """
+            )
+        start_check = any((start is None, start in df.columns))
+        stop_check = any((stop is None, stop in df.columns))
+        if not start_check:
+            raise ValueError(
+                """
+                The start value for the slice must either be `None`
+                or exist in the dataframe's columns.
+                """
+            )
+        if not stop_check:
+            raise ValueError(
+                """
+                The stop value for the slice must either be `None`
+                or exist in the dataframe's columns.
+                """
+            )
+        filtered_column = df.columns.slice_locs(start=start, end=stop)
+        # slice_locs fails when step has a value
+        # so this extra step is necessary to get the correct output
+        start, stop = filtered_column
+        filtered_column = df.columns[slice(start, stop, step)]
+
+    elif callable(columns_to_select):
+        # the function will be applied per series.
+        # this allows filtration based on the contents of the series
+        # or based on the name of the series,
+        # which happens to be a column name as well.
+        # whatever the case may be,
+        # the returned values should be a sequence of booleans,
+        # with at least one True.
+
+        filtered_column = [
+            columns_to_select(column) for _, column in df.items()
+        ]
+
+        # returns numpy bool, which does not work the same way as python's bool
+        # as such, cant use isinstance. pandas type check function helps out
+        checks = [pd.api.types.is_bool(column) for column in filtered_column]
+
+        if not all(checks):
+            raise ValueError(
+                "The callable provided must return a sequence of booleans."
+            )
+
+        # cant use `is` here, since it may be a numpy bool
+        checks = any(
+            (column == True for column in filtered_column)  # noqa: E712
+        )
+
+        if checks is False:
```

Comment:
Could be rewritten as `if not checks`.

Maybe something to think about for code readability purposes - is `passes_checks` possibly a more idiomatic variable name? We would get:

```python
if not passes_checks:
```